src/main.c

```diff
@@ -1343,14 +1351,28 @@ static void socket_reuseport(int fd)
 /**
  * Opens an INET or INET6 socket for accepting connections. When the protocol is UDP, SO_REUSEPORT is set if available.
  */
-static int open_listener(int domain, int type, int protocol, struct sockaddr *addr, socklen_t addrlen)
+static int open_listener(int domain, int type, int protocol, struct sockaddr *addr, socklen_t addrlen, int timestamping)
 {
     int fd;
 
     if ((fd = socket(domain, type, protocol)) == -1)
         goto Error;
     set_cloexec(fd);
 
+   {
+        int flag = 1;
+
+        if (timestamping) {
+#if defined(__linux__)
+            if (setsockopt(fd, SOL_SOCKET, SO_TIMESTAMPNS, &flag, sizeof(flag)) != 0) {
+                goto Error;
+            }
+#else /* !defined(__linux__) */
+            fprintf(stderr, "[warning] cannot set SO_TIMESTAMPNS because the platform does not support it\n");
+#endif /* defined(__linux__) */
```

Comment:
I like this approach (of emitting an warning on platforms that do not have support), and that coinsides with the idea of considering the concept of timestamping as platform-agnostic (please see my comment above).